cfme/tests/services/test_generic_service_catalogs.py

```diff
@@ -43,14 +43,14 @@ def test_delete_catalog_deletes_service(dialog, catalog):
     catalog.delete()
     service_catalogs = ServiceCatalogs("service_name")
     with error.expected(CandidateNotFound):
-        service_catalogs.order(catalog.name, catalog_item)
+        service_catalogs.order()
```

Comment:
You raised this exception handling as an area of concern in an email.

From what I'm seeing of how this was handled for catalog_item, the default exception raised by the table's select/click method is `NoSuchElementException`. The test yeild_fixture then catch and pass this exception when attempting to delete after yielding.

You can either modify your destinations to catch NoSuchElementException thrown by the table's click_row methods and re-raise as CandidateNotFound, or you can change the test here to expect a NoSuchElementException instead. 

CandidateNotFound exception was written specifically for accordion tree navigation failure, which is no longer the case here since you're navigating by the table and not by the accordion. That was likely the reason for not re-raising in catalog_item and just allowing the NoSuchElementException to pass through.